src/standalone/standard.rs

```diff
@@ -80,16 +83,43 @@ impl StandaloneStandardClient {
         tenant_id: &TenantId,
     ) -> Result<EncryptionKey, AlloyError> {
         let (key_id, edek_bytes) = Self::decompose_key_id_header(edek.0)?;
-        let secret = config_secrets.get(&key_id.0).ok_or_else(|| {
-            AlloyError::InvalidConfiguration(format!(
-                "Provided secret id `{}` does not exist in the standard configuration.",
-                &key_id.0
-            ))
-        })?;
-        let per_tenant_kek = derive_aes_encryption_key(&secret.secret, tenant_id);
-        let v4_document = Message::parse_from_bytes(&edek_bytes[..])
-            .map_err(|e| AlloyError::DecryptError(e.to_string()))?;
-        decrypt_aes_edek(&per_tenant_kek, &v4_document)
+
+        // Key id should never be 0. If it is we'll try all the keys we have.
+        let secrets = if key_id.0 == 0 {
+            config_secrets.values().collect_vec()
+        } else {
+            let secret = config_secrets.get(&key_id.0).ok_or_else(|| {
+                AlloyError::InvalidConfiguration(format!(
+                    "Provided secret id `{}` does not exist in the standard configuration.",
+                    &key_id.0
+                ))
+            })?;
+            vec![secret]
+        };
+
+        let attempt_decrypt = |secret: &Secret| {
+            let v4_document = crate::util::v4_proto_from_bytes(&edek_bytes)?;
+            // For standalone we want to try out the normal derivation first.
+            decrypt_aes_edek(
+                &derive_aes_encryption_key(&secret.secret, tenant_id),
+                &v4_document,
+            )
+            .or_else(|_| {
+                //If normal derivation doesn't work, we'll try our legacy derivation that was used in cloaked search.
+                decrypt_aes_edek(
+                    &derive_aes_encryption_key_legacy(&secret.secret, tenant_id),
+                    &v4_document,
+                )
+            })
+        };
+
+        secrets
+            .into_iter()
+            .map(attempt_decrypt)
+            .find_or_first(|result| result.is_ok())
+            .unwrap_or(Err(AlloyError::InvalidConfiguration(
+                "No secret could be found to decrypt".to_string(),
+            )))
```

Comment:
Cool find. I don't think the `unwrap_or` is possible, but I guess it's better to catch than expect?

Reply:
Ya, it shouldn't ever be able to happen, but if something went wrong I'd rather see an error than expect.